src/Other/Traits/TracksRevisions.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Rubix\ML\Other\Traits;
+
+use function hash;
+use function get_object_vars;
+use function implode;
+use function sort;
+
+/**
+ * Tracks Revisions
+ *
+ * @category    Machine Learning
+ * @package     Rubix/ML
+ * @author      Andrew DalPino
+ */
+trait TracksRevisions
+{
+    /**
+     * The revision number of the class.
+     *
+     * @return string
+     */
+    public function revision() : string
+    {
+        $properties = array_keys(get_object_vars($this));
```

Comment:
what if the type changes? This will be more relevant with typed props in php7.4

Reply:
This method is not perfect, there's other limitations as well that I'm aware of

This is more of a POC for now ... we could hash the property names and the types, or all the data for that matter